### PR TITLE
Remove `nGram` and  `edgeNGram` token filter names

### PR DESCRIPTION
**Pull request by cbuescher**
In #30209 we deprecated the camel case `nGram` filter name in favour of `ngram` and
did the same for `edgeNGram` and `edge_ngram`. Using these names has been deprecated
since 6.4 and is issuing deprecation warnings since then.
I think we can remove these filters in 8.0. In a backport of this PR I would change what was a
dreprecation warning from 6.4. to an error starting with new indices created in 7.0.

### CODE REVIEW

File: docs/reference/analysis/tokenfilters/edgengram-tokenfilter.asciidoc [hunk 1]
```diff
@@ -1,9 +1,9 @@
 [[analysis-edgengram-tokenfilter]]
 === Edge NGram Token Filter
 
-A token filter of type `edgeNGram`.
+A token filter of type `edge_ngram`.
```
Comment (cbuescher) [inline docs/reference/analysis/tokenfilters/edgengram-tokenfilter.asciidoc]: These doc-changes should go back to 7.0 as well.

File: modules/analysis-common/src/test/java/org/elasticsearch/analysis/common/HighlighterWithAnalyzersTests.java [hunk 2]
```diff
@@ -81,7 +81,7 @@ public void testNgramHighlightingWithBrokenPositions() throws IOException {
                         .put("analysis.tokenizer.autocomplete.max_gram", 20)
                         .put("analysis.tokenizer.autocomplete.min_gram", 1)
                         .put("analysis.tokenizer.autocomplete.token_chars", "letter,digit")
-                        .put("analysis.tokenizer.autocomplete.type", "nGram")
+                        .put("analysis.tokenizer.autocomplete.type", "ngram")
```
Comment (cbuescher) [inline modules/analysis-common/src/test/java/org/elasticsearch/analysis/common/HighlighterWithAnalyzersTests.java]: This is about the tokenizer, not the filter, but I think we also shouldn't use the camel case version of that one anymore.

File: modules/analysis-common/src/test/resources/rest-api-spec/test/analysis-common/30_tokenizers.yml [hunk 3]
```diff
@@ -133,7 +101,7 @@
             text: "foobar"
             explain: true
             tokenizer:
-              type: nGram
+              type: ngram
```
Comment (cbuescher) [inline modules/analysis-common/src/test/resources/rest-api-spec/test/analysis-common/30_tokenizers.yml]: same here, this is about the tokenizer, but we shouldn't use the camel case name here regardless

**Review by jimczi — APPROVED** [review]
LGTM, can you add an entry in the breaking changes (you'll need to create one for 8.0) ?

**Comment by cbuescher** [comment]
> LGTM, can you add an entry in the breaking changes (you'll need to create one for 8.0) ?

Will do. I was wondering if we also need to start throwing errors starting with 7.0 when token filters are used via "nGram" or "edgeNGram" in new indices, so essentially throwing an error where since 6.4 we have issued the deprecation warning, or if we can remove the filter in 8.0 only on the basis of having deprecated it in 6.4? The problem with token filters is AFAIK we currently cannot easily throw errors on e.g. index creation really but only when the filter is used (e.g. an index or analyze operation). Wdyt?

**Comment by jimczi** [comment]
> I was wondering if we also need to start throwing errors starting with 7.0 when token filters are used via "nGram" or "edgeNGram" in new indices, so essentially throwing an error

+1, we should throw an error if the deprecated name is used on an index created in 7.0+.

**Pull request merged**